packages/ngx-deploy-npm/src/executors/deploy/engine/engine.ts

```diff
@@ -1,8 +1,38 @@
 import { logger } from '@nx/devkit';
+import { readFileSync } from 'fs';
 
 import { setPackageVersion, NpmPublishOptions, spawnAsync } from '../utils';
 import { DeployExecutorOptions } from '../schema';
 
+async function checkIfPackageExists(
+  packageName: string,
+  version: string,
+  npmOptions: NpmPublishOptions
+): Promise<boolean> {
+  try {
+    const args = ['view', `${packageName}@${version}`, 'version'];
+    if (npmOptions.registry) {
+      args.push('--registry', npmOptions.registry);
+    }
+    await spawnAsync('npm', args);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function getPackageInfo(
+  distFolderPath: string
+): Promise<{ name: string; version: string }> {
+  const packageJson = JSON.parse(
+    readFileSync(`${distFolderPath}/package.json`, 'utf-8')
```

Comment:
We avoid using synchronous functions in the project to prevent blocking the Node.js main thread.

This will negatively impact the performance when running the target in parallel for large monorepos (`nx affected --target deploy`).

Please use an `async` function. There is an existing one on the project under `utils` folder called `readFileAsync`.

Reply:
Hi, thanks for your feedback! I have changed to readfileAsync